components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -116,6 +116,117 @@ std::string URLMethodToRequestType(ledger::type::UrlMethod method) {
   }
 }
 
+bool ResetOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const std::vector<std::string>& paths) {
+  bool result = true;
+
+  for (const auto& path : paths) {
+    const base::FilePath file_path =
+        base::FilePath::FromUTF8Unsafe(path.c_str());
+    if (!base::DeletePathRecursively(file_path)) {
+      result = false;
+    }
+  }
+
+  if (diagnostic_log) {
+    diagnostic_log->Close();
+  }
+
+  return result;
+}
+
+bool DeleteDiagnosticLogOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const std::string& path) {
+  if (diagnostic_log) {
+    diagnostic_log->Close();
+  }
+
+  const base::FilePath file_path = base::FilePath::FromUTF8Unsafe(path.c_str());
+  if (!base::PathExists(file_path)) {
+    return true;
+  }
+
+  return base::DeleteFile(file_path);
+}
+
+std::string LoadDiagnosticLogOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const std::string& log_path,
+    const int num_lines) {
+  if (!diagnostic_log) {
+    return std::string();
+  }
+
+  const base::FilePath file_path =
+      base::FilePath::FromUTF8Unsafe(log_path.c_str());
+  if (!base::PathExists(file_path)) {
+    return std::string();
+  }
+
+  std::string value;
+  if (!TailFileAsString(diagnostic_log.get(), num_lines, &value)) {
+    return base::StringPrintf(
+        "ERROR: %s",
+        GetLastFileError(diagnostic_log.get()).c_str());
+  }
+
+  return value;
+}
+
+bool MaybeTailDiagnosticLog(
+    base::WeakPtr<base::File> diagnostic_log,
+    const int num_lines) {
+  if (!diagnostic_log || !diagnostic_log->IsValid()) {
+    return false;
+  }
+
+  static bool first_run = true;
+
+  const int64_t length = diagnostic_log->GetLength();
+  if (length == -1) {
+    return false;
+  }
+
+  if (!first_run && length <= kDiagnosticLogMaxFileSize) {
+    return true;
+  }
+
+  first_run = false;
+
+  if (!TailFile(diagnostic_log.get(), num_lines)) {
+    return false;
+  }
+
+  return true;
+}
+
+bool WriteToDiagnosticLogOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const base::FilePath& log_path,
+    const std::string& log_entry,
+    const int num_lines) {
+  if (!InitializeLog(diagnostic_log.get(), log_path)) {
```

Comment:
Should we have a check (or `DCHECK`) here for `!diagnostic_log`?

Reply:
Yes, fixed.